File: C#/Leetcode/BinaryTree/SymmetricTree.cs
using LeetcodeSolutions.DataStructures;
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetcodeSolutions.BinaryTree
{
    // Leetcode 101 - https://leetcode.com/problems/symmetric-tree/
    // Submission Detail - https://leetcode.com/submissions/detail/171708245/

    public class SymmetricTree
    {
        public bool IsSymmetric(BinaryTreeNode root)
        {
            if (root == null)
                return true;

            return IsSymmetric(root.Left, root.Right);
        }

        public bool IsSymmetric(BinaryTreeNode root1, BinaryTreeNode root2)
        {
            if (root1 == null && root2 == null) return true;
            else if (root1 == null || root2 == null || root1.Val != root2.Val) return false;

            return IsSymmetric(root1.Left, root2.Right) && IsSymmetric(root1.Right, root2.Left);

            //bool leftRight = IsSymmetric(root1.Left, root2.Right);
            //bool rightLeft = IsSymmetric(root1.Right, root2.Left);

            //return leftRight && rightLeft;
        }
    }
}

File: Java/DFS/Tree/SymmetricTree.java
package DFS.Tree;

import DataStructures.TreeNode;

public class SymmetricTree {
    // Leetcode 101: https://leetcode.com/problems/symmetric-tree/description/
    // Submission Detail: https://leetcode.com/problems/symmetric-tree/submissions/1675598328

    //Input: root = [1,2,2,3,4,4,3]
    //Output: true

    //Input: root = [1,2,2,null,3,null,3]
    //Output: false

    public boolean isSymmetric(TreeNode root) {
        return isSymmetric(root.left, root.right);
    }

    private boolean isSymmetric(TreeNode node1, TreeNode node2) {
        // Base Case: Could be leaf node
        if(node1 == null && node2 == null)
            return true;

        // Base Case: nodes are not symmetric
        if(node1 == null || node2 == null || node1.val != node2.val)
            return false;

        return isSymmetric(node1.left, node2.right) && isSymmetric(node1.right, node2.left);
    }
}
